npm/ng-packs/packages/oauth/src/lib/guards/oauth.guard.ts

```diff
@@ -53,3 +53,33 @@ export const abpOAuthGuard: CanActivateFn = (
   authService.navigateToLogin(params);
   return false;
 };
+
+export const asyncAbpOAuthGuard: CanActivateFn = (
+  route: ActivatedRouteSnapshot,
+  state: RouterStateSnapshot,
+) => {
+  const authService = inject(AuthService);
+  const code = new URLSearchParams(window.location.search).get('code');
```

Comment:
We can use the response type to check whether it is coming from the server since the password flow does not have the same problem:

```tsx
  const { oAuthConfig } = environmentService.getEnvironment();

  if (oAuthConfig?.responseType === 'code') {
    //...
  }
```